### ISSUE TITLE
Project abandoned?

### ISSUE THREAD

**Issue by bj00rn** [thomasnordquist/MQTT-Explorer#730]
Thanks for this awesome app!

Is this project abandoned?

A lot of open issues and PRs.

**Comment by ilgrank** [comment]
Yes, the author is not updating the code since years and basically is also not very active here, his last activity has been a reply on this issue:
https://github.com/thomasnordquist/MQTT-Explorer/issues/683#issuecomment-1357485710
on Dec 19, 2022.

**Comment by bj00rn** [comment]
@ilgrank thanks, sad to hear..

@thomasnordquist would you consider loosening the license for this awesome app, so people can continue maintaining it?

**Comment by backupandforkacc** [comment]
Anyone with a good alternative to the software?

**Comment by XtremeOwnageDotCom** [comment]
> Anyone with a good alternative to the software?

Despite being unmaintained for a few years at this point, It's still arguably one of the best solutions out there.

**Comment by bj00rn** [comment]
> @ilgrank thanks, sad to hear..
> 
> @thomasnordquist would you consider loosening the license for this awesome app, so people can continue maintaining it?

PING @thomasnordquist, sorry for being a pain in the ***. Just can't wrap my head around this amazing piece of tech being unmaintained!

**Comment by jeeftor** [comment]
I suspect at some point somebody will just fork it and move it forward...

**Comment by adamalfath** [comment]
> Anyone with a good alternative to the software?

MQTTX is quite good

**Comment by jeeftor** [comment]
I still prefer this one unfortunately

**Comment by ilgrank** [comment]
@jeeftor : no update from @thomasnordquist  or @snordquist , right?

**Comment by jeeftor** [comment]
Of course no update

**Comment by XtremeOwnageDotCom** [comment]
> Somebody just needs to fork this and the community will likely jump in to help

The issue, is the license. 

https://github.com/thomasnordquist/MQTT-Explorer?tab=License-1-ov-file#readme

https://www.tldrlegal.com/license/creative-commons-attribution-noderivatives-4-0-international-cc-by-nd-4-0

Long story short, the license prevents others from modifying, and then sharing this code/project.

**Comment by XtremeOwnageDotCom** [comment]
So, unless @thomasnordquist changes to a less restrictive license, or decides to start maintaining this project again- 

It is as good as dead.

**Comment by jeeftor** [comment]
Doesn't the license only come into play if it's enforced? So if it's abandoned who is going to enforce it

**Comment by bj00rn** [comment]
> Doesn't the license only come into play if it's enforced? So if it's abandoned who is going to enforce it

That's not how licenses works though.. The Non-derivative license prevents us from forking and sharing the fork.

Beside the potentially legal exposure (however unlikely) putting effort into a project without a proper license is a no-go.

**Comment by jeeftor** [comment]
So what if the maintainer died? (Hypothetically)

**Comment by Jackfritt** [comment]
I think his family took it over to this one...
https://mqtt-explorer.com/
https://github.com/thomasnordquist/MQTT-Explorer

**Comment by MartinRiddar** [comment]
@jeeftor He isn't dead, he's just not very active on github anymore, he did a single pull request in the last year (November).
@Jackfritt If you check the Github link for that project it links here...

**Comment by scubanarc** [comment]
> > Doesn't the license only come into play if it's enforced? So if it's abandoned who is going to enforce it
> 
> That's not how licenses works though.. The Non-derivative license prevents us from forking and sharing the fork.
> 
> Beside the potentially legal exposure (however unlikely) putting effort into a project without a proper license is a no-go.

If you rewind the clock 5 years, it was AGPL:

https://github.com/thomasnordquist/MQTT-Explorer/tree/ea7994dad945dba2291d097ea233def9942761e3?tab=readme-ov-file

**Comment by XtremeOwnageDotCom** [comment]
> If you rewind the clock 5 years, it was AGPL:
> 
> https://github.com/thomasnordquist/MQTT-Explorer/tree/ea7994dad945dba2291d097ea233def9942761e3?tab=readme-ov-file

Doesn't look like there are that many differences in the last 5 years. Perhaps we should fork that to a new project.

**Comment by scubanarc** [comment]
Yes, someone should fork it from 5 years ago when it was still AGPL, everyone move their PR's, and then we all continue on forward with this project.

**Comment by steveja** [comment]
Maybe it's me - but I can't build from the git repo.  'yarn' complains that the older packages are needed (my interp').     So I think this is totally dead as an open project.

**Comment by jeeftor** [comment]
So the license got added in this commit (2019).

https://github.com/thomasnordquist/MQTT-Explorer/commit/9320fa252e0a630f3d933fbb3bcd83847ca6d381

- Does that mean anything prior to this commit is fair game?

**Comment by XtremeOwnageDotCom** [comment]
> So the license got added in this commit (2019).
> 
> [9320fa2](https://github.com/thomasnordquist/MQTT-Explorer/commit/9320fa252e0a630f3d933fbb3bcd83847ca6d381)
> 
> * Does that mean anything prior to this commit is fair game?

Basically, up to the limitations of its prior license

**Comment by thomasnordquist** [comment]
Hello there,
for me time is a very limited resource, and at the end of the day I try to finally close my laptop.
This is one reason I do not maintain this project (or any other private projects) at the moment.

The license will stay like this for now, the main goal is to protect this project to be commercially distributed in any way.

My previous experiences with collborators was sadly not optimal, therefore I was discouraged to pursue this further.
I will request @bj00rn to be a maintainer of this project (since he already spend some effort to get in contact with me).

I will also help with the distribution of new version.

There are a few things which are kind of urgent for this project:
 - A new version of electron (outdate SSL certificates)
 - Removal of the telemetry (not used anymore / opt-in telemetry)
   - since it is no longer operational removing it should be easy to do
   - option: self-hosted telemetry / vs. telemetry service free for open source usage

**Comment by scubanarc** [comment]
Hey Thomas, glad to hear you're still around! Thanks again for this awesome program. It really is the best MQTT frontend out there and I use it daily. Amazing work!

> The license will stay like this for now, the main goal is to protect this project to be commercially distributed in any way.

That's very unfortunate. Please reconsider this.

It was AGPL once before, so it's already out there to be abused by companies. Keeping it [CC-BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0/) just hurts the community by preventing an active user from forking and improving it.

As of right now, it's been dead for 2 years. There are 64 PRs and no-one to merge them.

I do hope that @bj00rn is able to maintain it, but please consider opening up the license.

**Comment by GeekyTim** [comment]
@thomasnordquist, I'm sorry to hear you have been burnt by previous collaborations, and understand your stance. Please remember that many of us here in this community don't always have the best communication methods and often cannot see other's reasons and point of view. Being recently diagnosed as Autistic myself, I do know I have not always been the best communicator in the past.

I will not comment on your choice of license etc. That is your choice alone.

@bj00rn, I welcome any work you do on this tool. I am looking forward in particular to see it working on the Raspberry Pi, my current main machine!

**Comment by jeeftor** [comment]
Is there a license which prohibits commercial use but opens up for open source - forking etc? Or does such a thing not exist?

**Comment by XtremeOwnageDotCom** [comment]
> Is there a license which prohibits commercial use but opens up for open source - forking etc? Or does such a thing not exist?

Creative Commons NonCommercial

**Comment by bj00rn** [comment]
@thomasnordquist thanks for the invite! I wasn't really expecting to be assigned maintainer but merely hoping to re-open a discussion on how to move this awesome project forward. That being said, I am willing to invest some time and effort to get this project up to date and try to get some momentum going here.

Let's get the janitorial side of things up to date first so we can do a new release. 
- Dependencies are way outdated
- CI is failing. 

Additionally there are a few long due, low hanging PRs that seem like they could be merged without any major considerations. 

*Disclaimer*
Im pretty well versed in react+redux but my skills concerning electron are basic at the very best, so please be kind/patient here.

If some of you want to contribute I feel free to open PRs. This will have to be a team effort going forward.

**Comment by micw** [comment]
Great to see that the project is not abandoned!

I was able to fix the certificate issue locally by upgrading electron to the latest release (29.0.1). I also updated electron-builder to 24.12.0, no idea if that matters. To build with a recent node version, I also added `export NODE_OPTIONS=--openssl-legacy-provider &&` in front of the commands in the "build" script definition.

Important was to start from git tag 355e9177fc1be24bd1eb516772aede3877a7de32 which was the tag the latest beta-release was made from. The changes after caused several issues. So maybe branching from there would make sense for a hotfix release.

**Comment by bj00rn** [comment]
Please have a look at https://github.com/thomasnordquist/MQTT-Explorer/pull/757

**Comment by nicklasb** [comment]
@thomasnordquist , @bj00rn :
Let me just say that is just GREAT!
And yeah, collaboration can be hard, especially in open source, sometimes the giving and the taking doesn't add up at all.

**Comment by bj00rn** [comment]
@thomasnordquist Can you please have a look at master, It should be ready for a pre-release.

**Comment by thomasnordquist** [comment]
Thank you, I will work on it this week.

**Comment by bj00rn** [comment]
> Thank you, I will work on it this week.

Give me a shout if something is broken.

Some notes:

* Packaging: I've only tried building/packaging locally since i have no knowledge of the CI inner workings.

* "E2E Tests": I had to replace spectron with playwright since spectron is deprecated and lacks newer electron support. I had to disable some steps in the video/test script because I couldn't figure out how to correctly expand the correct subnodes in the tree (playwright strictly follows accessibility rules and won't allow click events on elements that are not visible).  I.e for `Livingroom/Lamp` to be clickable, `Livingroom` must first be clicked. I made some progress, but currently it does not work with non-unique node Names, hence the disabled tests.

https://github.com/thomasnordquist/MQTT-Explorer/blob/737afb3c1b9f58aa9ff639a0e91ed86d2cd2b5c9/app/src/components/Tree/TreeNode/TreeNodeTitle.tsx#L22
https://github.com/thomasnordquist/MQTT-Explorer/blob/737afb3c1b9f58aa9ff639a0e91ed86d2cd2b5c9/src/spec/demoVideo.ts#L87
https://github.com/thomasnordquist/MQTT-Explorer/blob/737afb3c1b9f58aa9ff639a0e91ed86d2cd2b5c9/src/spec/util/expandTopic.ts#L4


* The sparkplug_b protocol file has been moved from `res` folder to a JS module that exports the protocol as a string, I couldn't figure out how to get the `.proto` file to be included in packaging.
https://github.com/thomasnordquist/MQTT-Explorer/blob/737afb3c1b9f58aa9ff639a0e91ed86d2cd2b5c9/backend/src/Model/sparkplugb.proto.ts#L1

**Comment by samfweb** [comment]
@bj00rn, @backupandforkacc, @jeeftor and others.

I built a true open-source alternative (AGPL) to this from scratch because of the issues you've highlighted above.

Check out https://github.com/mqtt-viewer/mqtt-viewer!

![Image](https://github.com/user-attachments/assets/7e1ec72a-a4f9-41b4-b7c9-4ab31c82928c)

**Comment by samfweb** [comment]
@luebbe Yea I can see how it would be annoying—I wouldn't have done it if this project was actively maintained. 

However, the majority of issues I responded to are problems that MQTT Viewer solves and that very likely won't be solved in this project. I don't really see another way of letting those guys know there's something out there that might help them.

**Comment by thomasnordquist** [comment]
@samfweb I wish you luck with your tool, but the spam is quite annoying
